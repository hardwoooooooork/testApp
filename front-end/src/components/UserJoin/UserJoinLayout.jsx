import React from 'react'
import { Button, Header, Icon, Modal, Form } from 'semantic-ui-react'


export default function UserJoinLayout(prop) {
    
    return (
        <Modal
            trigger={<span onClick={prop.handleOpen} style={{ "color": "blue", "marginRight": "7px", "cursor": "pointer" }}>Sign Up</span>}
            open={prop.modalOpen}
            onClose={prop.handleClose}
            size='small'
        >
            <Header icon='browser' content='User Join' />
            <Modal.Content>
                <Form>
                    <Form.Field>
                        <label>ID</label>
                        <input type="text" name="userId" placeholder='User id' onChange={prop.onChangeHandler} value={(prop.formData.userId) ? prop.formData.userId: ''} />
                    </Form.Field>
                    <Form.Field>
                        <label>Name</label>
                        <input type="text" name="userName" placeholder='User Name' onChange={prop.onChangeHandler} value={(prop.formData.userName) ? prop.formData.userName: ''} />
                    </Form.Field>
                    <Form.Field>
                        <label>Password</label>
                        <input type="password" name="userPassword" placeholder='User Password' onChange={prop.onChangeHandler} value={(prop.formData.userPassword) ? prop.formData.userPassword: ''} />
                    </Form.Field>
                </Form>
            </Modal.Content>
            <Modal.Actions>
                <Button color='green' type='button' inverted onClick={prop.onsubmitHandler}>
                    <Icon name='checkmark' /> Submit</Button>
                <Button color='red' onClick={prop.handleClose} inverted>
                    <Icon name='x' /> Close
          </Button>
            </Modal.Actions>
        </Modal>
    )
}
