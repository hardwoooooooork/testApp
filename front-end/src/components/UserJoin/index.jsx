import React, { Component } from 'react'
import UserJoinLayout from './UserJoinLayout';
import * as user from '../../services/login'

export default class UserJoin extends Component {

    state = { modalOpen: false, formData:{} }

    handleOpen = () => this.setState({ modalOpen: true })

    handleClose = () => this.setState({ modalOpen: false })
    
    onChangeHandler = (e) => this.setState({ formData: { ...this.state.formData, [e.target.name]: e.target.value } })

    onsubmitHandler = async () => {
        try {
            const res = await user.userJoin(this.state.formData);
            //console.log(res);
            
            if (res.status === 200) {
                alert("가입이 완료되었습니다.")
                this.setState({ modalOpen: false, formData: {} })
            }
        } catch (error) {
            alert("가입 에러 발생")
        }
        
    }
    render() {
        return (
            <UserJoinLayout
                handleOpen={this.handleOpen}
                handleClose={this.handleClose}
                onChangeHandler={this.onChangeHandler}
                onsubmitHandler={this.onsubmitHandler}
                {...this.state}
            />
        )
    }
}
