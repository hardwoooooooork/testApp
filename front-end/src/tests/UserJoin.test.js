import React from 'react';
import { render } from '@testing-library/react';
import UserJoin from '../components/UserJoin';


describe('<UserJoin />', () => {
    it('matches snapshot', () => {
        const utils = render(<UserJoin />);
        expect(utils.container).toMatchSnapshot();
    });
});