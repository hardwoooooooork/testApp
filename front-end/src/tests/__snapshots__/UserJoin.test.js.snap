// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserJoin /> matches snapshot 1`] = `
<div>
  <span
    style="color: blue; margin-right: 7px; cursor: pointer;"
  >
    Sign Up
  </span>
</div>
`;
